src/main/java/com/devotedmc/ExilePearl/listener/PlayerListener.java

```diff
@@ -424,6 +446,20 @@ public void onPlayerPickupPearl(EntityPickupItemEvent e) {
 		}
 	}
 
+	/**
+	 * Listen for pearl being moved to item frame
+	 * @param event
+	 */
+	@EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
+	public void onPlayerInteractEntity(PlayerInteractEntityEvent event) {
```

Comment:
Is there some other existing listener that handles the item frame being broken or the pearl being removed or how does that work?

Reply:
Anytime the pearl leaves the item frame, `ItemSpawnEvent`  should be [called](https://github.com/CivClassic/ExilePearl/blob/c02223f95968b9db48861156d9db8adf56db84b8/src/main/java/com/devotedmc/ExilePearl/listener/PlayerListener.java#L187).

Reply:
Ah, neat